src/pf-tooltip/index.html

```diff
@@ -0,0 +1,51 @@
+<!doctype html>
+<html>
+<head>
+    <title>pf-tooltip example</title>
+    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/patternfly/3.13.0/css/patternfly-additions.css">
+    <link rel="stylesheet" href="/dist/css/patternfly.css">
+    <script src="//cdnjs.cloudflare.com/ajax/libs/webcomponentsjs/0.7.22/webcomponents.js"></script>
+    <script src="/dist/js/pf-utils.js"></script>
+    <script src="/dist/js/pf-tooltip.js"></script>
+</head>
+<body>
+<div class="container">
+    <div class="page-header">
+        <h1>Tooltips</h1>
+    </div>
+    <hr>
+
+    <pf-tooltip placement="left" targetSelector="#btn-left">
+        Tooltip text on Left
+    </pf-tooltip>
+    <button id="btn-left" type="button" class="btn btn-primary">Left</button>
+
+    <pf-tooltip placement="top" targetSelector="#btn-top">
+        <span>Tooltip on top</span>
+    </pf-tooltip>
+    <button id="btn-top" type="button" class="btn btn-secondary">Top</button>
+
+    <pf-tooltip placement="bottom" targetSelector="#btn-bottom">
+        <span>Tooltip on bottom</span>
+    </pf-tooltip>
+    <button id="btn-bottom" type="button" class="btn btn-warning">Bottom</button>
+
+    <pf-tooltip placement="right" targetSelector="#btn-right">
+        <span>Tooltip on right</span>
+    </pf-tooltip>
+    <button id="btn-right" type="button" class="btn btn-danger">Right</button>
+
+    <br/>
+    <br/>
+    <script>
+        setTimeout(function(){
+          var pfTooltipRight = document.querySelector('[targetSelector="#btn-right"]');
+          pfTooltipRight.innerHTML = "Right content changed!";
+
+          var event = new CustomEvent('handleContentChanged',{});
+          pfTooltipRight.dispatchEvent(event);
```

Comment:
This is a really important point (and one of the main things I was questioning on this and other similar components). I have talked with other devs about this I'd consider pretty knowledgeable in the WC space too and I suppose this is my preferred option right now (not saying we have to use it globally, but I think it works well here and is the most optimized pattern right now). MutationObservers can be quite slow if overused and events should be able to process outside framework changes more directly. Also, I think events are closer to the eventual pattern we'd be able to use with "slotchanged" once we implement Shadow DOM.  I've noticed this is a pretty common pattern in the WC space.
https://github.com/webcomponents/gold-standard/wiki/Content-Changes

My thinking is that we could write similar shim in Angular that I have done for React, and have the directive shim observe attribute changes and then fire the event to notify the WC. This way when scope variables/attributes change, we can abstract the actual event logic away in the shim, and the consumer doesn't need to know about it.

I  am hoping to write up an example of this today.